modules/claims_api/app/controllers/claims_api/v1/application_controller.rb

```diff
@@ -51,7 +51,7 @@ def validate_veteran_identifiers(require_birls: false) # rubocop:disable Metrics
 
         if target_veteran.mpi.mpi_response_is_cached?
           target_veteran.mpi.destroy
-          target_veteran.mpi = nil
+          target_veteran.mpi = nil unless target_veteran.mpi.nil?
```

Comment:
I'm not sure the scenario we would see this. we don't have a setter on that class (ClaimsApi::Veteran for mpi), that's why it's complaining.
```
    def mpi
      @mpi ||= MPIData.for_user(self)
    end
```

that being said, the hacky way to do this is the `attr_accessor :mpi` or `attr_writer` which would at least bypass the no method error. I don't think that's our solution either though, I'd like to better understand when we hit this scenario, and what the `target_veteran` looks like. Do we have any context on the request coming in or when we'd hit this? Maybe some details of why we set this up in the first place would be helpful.

Reply:
history: https://github.com/department-of-veterans-affairs/vets-api/pull/10674